fix linting issues

thaJeztah opened the pull request:
- relates to https://github.com/microsoft/go-winio/pull/324#issuecomment-2584758696

thaJeztah committed:
fix linting issues

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
cc @kevpar if  you have permissions to kick CI ❤️

katiewasnothere commented:
@thaJeztah It looks like the CI is still failing

thaJeztah commented:
Yeah, I noticed; I suspect the other failures may also be with newer go versions, or updated golangci-lint.

I think they're all about integer conversion in generated code; haven't checked if they can be ignored (temporarily) or need fixing;

```
mode=readonly] in [] ...
  ::high file=wim/lzx/lzx.go,line=137,col=13::G115: integer overflow conversion uint32 -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=191,col=26::G115: integer overflow conversion uint -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=200,col=38::G115: integer overflow conversion int -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=376,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=377,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=378,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
  ::high file=wim/lzx/lzx.go,line=526,col=51::G115: integer overflow conversion int -> uint16 (gosec)

kiashok reviewed (changes requested):
CI is still failing

kiashok commented:
@mat007 @thaJeztah looks like the CIs are still failing

mat007 commented:
Maybe these should be pinned version? 🤷 
https://github.com/microsoft/go-winio/blob/bdc6c112805ce13d7b235dc6257f258d958f5471/.github/workflows/ci.yml#L7-L9

thaJeztah committed:
golangci: temporarily ignore G115 (gosec)

It currently produces too many (false) positives in generated code, which
needs to be reviewed. For the time being, exclude this check.

    ::high file=wim/lzx/lzx.go,line=137,col=13::G115: integer overflow conversion uint32 -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=191,col=26::G115: integer overflow conversion uint -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=200,col=38::G115: integer overflow conversion int -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=376,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=377,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=378,col=20::G115: integer overflow conversion uint32 -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=526,col=51::G115: integer overflow conversion int -> uint16 (gosec)
    ::high file=wim/lzx/lzx.go,line=546,col=23::G115: integer overflow conversion int64 -> int32 (gosec)
    ::high file=wim/lzx/lzx.go,line=547,col=16::G115: integer overflow conversion uint32 -> int32 (gosec)
    ::high file=wim/lzx/lzx.go,line=555,col=53::G115: integer overflow conversion int32 -> uint32 (gosec)
    ::high file=wim/lzx/lzx.go,line=567,col=32::G115: integer overflow conversion int -> uint16 (gosec)
    ::high file=internal/stringbuffer/wstring.go,line=131,col=47::G115: integer overflow conversion int -> uint32 (gosec)
    ::high file=pkg/bindfilter/bind_filter.go,line=231,col=66::G115: integer overflow conversion int -> uint32 (gosec)
    ::high file=pkg/bindfilter/bind_filter.go,line=241,col=16::G115: integer overflow conversion int -> uint32 (gosec)
    ::high file=pkg/process/process.go,line=19,col=23::G115: integer overflow conversion int -> uint32 (gosec)

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented on .golangci.yml:
```diff
@@ -72,6 +72,9 @@ linters-settings:
   nolintlint:
     require-explanation: true
     require-specific: true
+  gosec:
+    excludes:
+      - G115 # G115: integer overflow conversion; TODO: currently too many (false) positives in generated code that need to be verified
```
I added a global ignore for this one for now; I think most of these are likely false positives in generated code, but would have to be reviewed by someone with more understanding of those parts of the code.

thaJeztah commented:
Pushed some changes, but I don't have a windows machine to verify on locally, but if someone could kick CI again ❤️  @kiashok @katiewasnothere

AdityaMittal1306 commented on hvsock_test.go:
```diff
@@ -619,7 +619,7 @@ func (u testUtil) WaitErr(ch <-chan error, d time.Duration, msgs ...string) {
 		if err != nil {
 			u.T.Helper()
-			u.T.Fatalf(msgJoin(msgs, "%v"), err)
+			u.T.Fatalf(msgJoin(msgs, err.Error()))
```
This should be Fatal, not Fatalf, to stay consistent with the Must fix below, any % in the error message would be misinterpreted as a format verb

AdityaMittal1306 commented on hvsock_test.go:
Looks like this timeout case was missed — it still uses the old Fatalf pattern while the identical case in Wait was fixed.

AdityaMittal1306 commented:
This PR has outstanding merge conflicts , can you pls check these @thaJeztah
